packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -178,18 +226,68 @@ class RebaseQueue<T> {
             const newMarkLength = getInputLength(newMark);
             const baseMarkLength = getInputLength(baseMark);
             if (newMarkLength < baseMarkLength) {
-                const [baseMark1, baseMark2] = splitMarkOnInput(baseMark, newMarkLength);
+                const [baseMark1, baseMark2] = splitMarkOnInput(
+                    baseMark,
+                    newMarkLength,
+                    this.genId,
+                    this.moveEffects,
+                );
                 this.baseMarks.push(baseMark2);
                 return { baseMark: baseMark1, newMark };
             } else if (newMarkLength > baseMarkLength) {
-                const [newMark1, newMark2] = splitMarkOnInput(newMark, baseMarkLength);
+                const [newMark1, newMark2] = splitMarkOnInput(
+                    newMark,
+                    baseMarkLength,
+                    this.genId,
+                    this.moveEffects,
+                );
                 this.newMarks.push(newMark2);
+                this.moveEffects.validatedMarks.add(newMark1);
+                this.moveEffects.validatedMarks.add(newMark2);
                 return { baseMark, newMark: newMark1 };
             } else {
                 return { baseMark, newMark };
             }
         }
     }
+
+    private getNextBaseMark(): Mark<T> | undefined {
+        return this.getNextMark(this.baseMarks, false, undefined);
+    }
+
+    private getNextNewMark(): Mark<T> | undefined {
+        return this.getNextMark(this.newMarks, true, undefined);
+    }
+
+    private getNextMark(
+        marks: StackyIterator<Mark<T>>,
+        reassignMoveIds: boolean,
+        revision: RevisionTag | undefined,
```

Comment:
This is always `undefined` in the current implementation. Is that by design?

Reply:
We should pass `this.baseRevision` in `getNextBaseMark`.